packages/viz/src/CartoTileLayer.ts

```diff
@@ -1,88 +1,84 @@
+    this._mapInstantiation = this._mapsClientInstance.instantiateMapFrom(
+      buildInstantiationOptions({ mapOptions: this._mapOptions, mapSource: this._mapSource })
+    );
+  }
 
-    this._deck = deckInstance || window.deck;
-    this._mapOptions = Object.assign({}, mapOptions, {
-      vector_extent: 2048,
-      vector_simplify_extent: 2048
+  public async buildDeckGLLayer(layerProps: { layerType?: any} = {}) {
+    // TODO: Parse through Babel
+    const { layerType: sublayerType, ...styleProps} = layerProps;
+    const deckSublayer = sublayerType || GeoJsonLayer;
+
+    const { urlTemplates } = await this._mapInstantiation.then(this._parseInstantiationResult);
+
+    const layerProperties = Object.assign({}, styleProps, {
+      getLineColor: [192, 0, 0],
+      getFillColor: [200, 120, 80],
+      lineWidthMinPixels: 1,
+      pointRadiusMinPixels: 5,
+      urlTemplates,
+      uniquePropertyName: 'cartodb_id',
+      renderSubLayers: (props: any) => new deckSublayer({ ...props })
     });
-    this._credentials = credentials || Credentials.default;
 
-    // this._style = style;
-    this._mapsClient = new Maps(this._credentials);
+    return new MVTTileLayer(layerProperties);
   }
 
-  public async buildDeckglLayer(props: { layerType?: any} = {}) {
-    // const layerType = props.layerType;
-    const styleProps = Object.assign({}, props, { layerType: undefined });
-
-    // const deckLayer = layerType || this._deck.GeoJsonLayer;
-
-    const urlTemplates = await this.instantiateMap().then((data: any) => {
-      const urlData = data.metadata.url.vector;
+  private _parseInstantiationResult(instantiationData: any) {
+    const urlData = instantiationData.metadata.url.vector;
 
-      return urlData.subdomains.map(
-        (subdomain: any) => urlData.urlTemplate.replace('{s}', subdomain)
-      );
-    });
-
-    return new this._deck.MVTTileLayer(
-      Object.assign({}, styleProps, {
-        getLineColor: [192, 0, 0],
-        getFillColor: [200, 120, 80],
-        lineWidthMinPixels: 1,
-        pointRadiusMinPixels: 5,
-        urlTemplates,
-        uniquePropertyName: 'cartodb_id',
-        // renderSubLayers: (props) => {
-        //   return new deckLayer({
-        //     ...props
-        //   });
-        // }
-      })
+    const urlTemplates = urlData.subdomains.map(
+      (subdomain: string) => urlData.urlTemplate.replace('{s}', subdomain)
     );
+
+    return { urlTemplates };
   }
+}
 
-  private async instantiateMap() {
-    let options;
-    if (this._source.split(' ').length === 1) {
-      options = Object.assign({}, this._mapOptions, { dataset: this._source });
-    } else {
-      options = Object.assign({}, this._mapOptions, { sql: this._source });
-    }
+function buildInstantiationOptions(
+  { mapOptions, mapSource }: { mapOptions: MapOptions, mapSource: Source}
+) {
+  return {
+    ...mapOptions,
+    ...mapSource.getSourceOptions()
```

Comment:
I see this more like an implementation detail of the Maps API. Should we consider moving these methods to the maps package itself?

Reply:
I guess so! I left them here because they were here before, but I agree that it makes more sense to move them there.

Reply:
I was rethinking this, and I'm not sure about this. If we move this function to Maps package, we should move our Source declaration there as well, so that the Maps package is aware of types.